vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj

```diff
@@ -807,6 +807,8 @@
     <PackageReference Include="Microsoft.Windows.CppWinRT" Version="$(CppWinRTVersion)" />
     <PackageReference Include="ReactNative.Hermes.Windows" Version="$(HermesVersion)" />
     <PackageReference Include="$(WinUIPackageName)" Version="$(WinUIPackageVersion)" />
+    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0" />
```

Comment:
I think this is the build issue:

When setting `UseWinUI3`, `WinUIPackageName` expands as `Microsoft.WindowsAppSDK`.
By invariantly adding it as a dependency, it "competes" with the default WinUI (2.7) hence producing the error.

I'll confirm with a test build and submit a fix suggestion if successful.

Reply:
I don't think we need the Microsoft.WindowsAppSDK explicit package here, if WinUI.props sets WinUIPackageName and WinUIPackageVersion to be the winappsdk pkg/version when using --useWinUI3; that was the original intent at least

Reply:
Not only we don't need it.
Because `PackageReference` implicitly processes packages' contained MSBuild logic, this caused an error when `WuinUIPackageName` was pointing to WinUI2.

I have verified and committed the fix (simply removing this line).

Reply:
Nice catch. Thank you, @JunielKatarn!